apps/server/src/app.ts

```diff
@@ -324,3 +330,46 @@ process.on('uncaughtException', async (error) => {
 process.once('SIGHUP', async () => shutdown(0));
 process.once('SIGINT', async () => shutdown(0));
 process.once('SIGTERM', async () => shutdown(0));
+
+/**
+ * @description tries to open the server with the desired port, and if getting a `EADDRINUSE` will change to an efemeral port
+ * @param {http.Server}server http server object
+ * @param {number}desiredPort the desired port
+ * @returns {number} the resulting port number
+ * @throws any other server errors will result in a throw
+ */
+async function serverTryDesiredPort(server: http.Server, desiredPort: number): Promise<number> {
+  return new Promise((res) => {
+    expressServer.once('error', (e) => {
+      if (isDocker) throw e; // we should only move ports if we are in a desktp environment
+      if (testForPortInUser(e)) {
+        server.listen(0, '0.0.0.0', () => {
+          // @ts-expect-error TODO: find proper documentation for this api
```

Comment:
can i help?

Reply:
ts dosn't like the type returned
[serveraddress](https://nodejs.org/api/net.html#serveraddress)
it might be just be a case of having to narrow it down

Reply:
tricky, from the docs, I am not sure what this means

> For a server listening on a pipe or Unix domain socket, the name is returned as a string.

so we have two options:
- a) create a type guard to check what the response is
- b) if we are sure that in our cases we always get the AddressInfo, we can just cast the response like

```ts
          const {port} = server.address() as AddressInfo;
```

b) could be fine, as if we get a string in the first example we would need to decide what to do